gobblin-yarn/src/main/java/gobblin/yarn/GobblinHelixTask.java

```diff
@@ -98,82 +93,97 @@ public GobblinHelixTask(TaskCallbackContext taskCallbackContext, TaskExecutor ta
 
   @Override
   public TaskResult run() {
-    LOGGER.info(String.format("Running task %s of job %s", taskId, jobId));
-
     try {
-      // Create a new task from the work unit and submit the task to run
-      gobblin.runtime.Task task = buildTask();
-      this.taskStateTracker.registerNewTask(task);
-      LOGGER.info(String.format("Submitting Task %s to run", this.taskId));
-      this.futureOptional = Optional.<Future<?>>of(this.taskExecutor.submit(task));
-      this.futureOptional.get().get();
-
-      // Persist the task state so it is later collected by the job launcher
-      persistTaskState(task.getTaskState());
-
-      WorkUnitState.WorkingState workingState = task.getTaskState().getWorkingState();
-      LOGGER.info(String.format("Task %s completed with state %s", this.taskId, workingState));
-
-      switch (workingState) {
-        case SUCCESSFUL:
-          return new TaskResult(TaskResult.Status.COMPLETED, "task id: " + this.taskId);
-        case FAILED:
-          return new TaskResult(TaskResult.Status.ERROR, "task id: " + this.taskId);
-        case CANCELLED:
-          return new TaskResult(TaskResult.Status.CANCELED, "task id: " + this.taskId);
-        default:
-          throw new IllegalStateException("Unexpected result task state: " + workingState);
+      Path workUnitFilePath =
```

Comment:
A lot of the code below seems similar to what is done in `MRJobLauncher.TaskRunner` anyway we can abstract the code to a new class so we can re-use it for both MR jobs and YARN jobs?